src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,965 @@
+                contentType);
+            WriteVerbose(respVerboseMsg);
+        }
+
+        private static string StringFormatCurrentCulture(string format, params object[] args)
+        {
+            return string.Format(CultureInfo.CurrentCulture, format, args);
+        }
+
+        private static string FormatDictionary(IDictionary content)
+        {
+            if (content == null)
+                throw new ArgumentNullException(nameof(content));
+
+            var bodyBuilder = new StringBuilder();
+            foreach (string key in content.Keys)
+            {
+                if (0 < bodyBuilder.Length) bodyBuilder.Append("&");
+
+                var value = content[key];
+
+                // URLEncode the key and value
+                var encodedKey = WebUtility.UrlEncode(key);
+                var encodedValue = string.Empty;
+                if (value != null) encodedValue = WebUtility.UrlEncode(value.ToString());
+
+                bodyBuilder.AppendFormat("{0}={1}", encodedKey, encodedValue);
+            }
+
+            return bodyBuilder.ToString();
+        }
+
+        private HttpRequestMessage GetRequest(Uri uri)
+        {
+            var requestUri = PrepareUri(uri);
+            var httpMethod = GetHttpMethod(Method);
+            // create the base WebRequest object
+            var request = new HttpRequestMessage(httpMethod, requestUri);
+
+            // pull in session data
+            if (GraphRequestSession.Headers.Count > 0)
+            {
+                GraphRequestSession.ContentHeaders.Clear();
+                foreach (var entry in GraphRequestSession.Headers)
+                {
+                    if (HttpKnownHeaderNames.ContentHeaders.Contains(entry.Key))
+                    {
+                        GraphRequestSession.ContentHeaders.Add(entry.Key, entry.Value);
+                    }
+                    else
+                    {
+                        if (SkipHeaderValidation)
+                        {
+                            request.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
+                        }
+                        else
+                        {
+                            request.Headers.Add(entry.Key, entry.Value);
+                        }
+                    }
+                }
+            }
+
+            // Set 'Transfer-Encoding: chunked' if 'Transfer-Encoding' is specified
+            if (GraphRequestSession.Headers.ContainsKey(HttpKnownHeaderNames.TransferEncoding))
+            {
+                request.Headers.TransferEncodingChunked = true;
+            }
+
+            // Set 'User-Agent' if WebSession.Headers doesn't already contain it
+            if (GraphRequestSession.Headers.TryGetValue(HttpKnownHeaderNames.UserAgent, out var userAgent))
+            {
+                GraphRequestSession.UserAgent = userAgent;
+            }
+            else
+            {
+                if (SkipHeaderValidation)
```

Comment:
`else if`

Reply:
Please elaborate...Looks okay to me. Is there something I am missing?